pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/WindowNode.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.stage;
+
+import com.clearspring.analytics.util.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.calcite.rel.RelFieldCollation;
+import org.apache.calcite.rel.core.Window;
+import org.apache.calcite.rex.RexLiteral;
+import org.apache.calcite.rex.RexWindowBound;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.query.planner.logical.RexExpression;
+import org.apache.pinot.query.planner.serde.ProtoProperties;
+
+
+public class WindowNode extends AbstractStageNode {
+  @ProtoProperties
+  public List<RexExpression> _groupSet;
```

Comment:
could you add a test to `SerDeUtilsTest` to ensure WindowNode is Ser/De properly?

Reply:
I did already, as I added a bunch of Window function queries to `QueryEnvironmentTestBase::provideQueries`. The `SerDeUtilsTest` uses the `dataProvider` as `QueryEnvironmentTestBase::provideQueries`.

```
public class SerDeUtilsTest extends QueryEnvironmentTestBase {

  @Test(dataProvider = "testQueryDataProvider")
  public void testQueryStagePlanSerDe(String query)
      throws Exception {
    QueryPlan queryPlan = _queryEnvironment.planQuery(query);
    for (StageNode stageNode : queryPlan.getQueryStageMap().values()) {
      Plan.StageNode serializedStageNode = StageNodeSerDeUtils.serializeStageNode((AbstractStageNode) stageNode);
      StageNode deserializedStageNode = StageNodeSerDeUtils.deserializeStageNode(serializedStageNode);
      Assert.assertTrue(isObjectEqual(stageNode, deserializedStageNode));
      Assert.assertEquals(deserializedStageNode.getStageId(), stageNode.getStageId());
      Assert.assertEquals(deserializedStageNode.getDataSchema(), stageNode.getDataSchema());
      Assert.assertEquals(deserializedStageNode.getInputs().size(), stageNode.getInputs().size());
    }
  }
...
...
}
```